docs/en/sql-reference/functions/string-replace-functions.md

```diff
@@ -248,3 +248,75 @@ select printf('%%%s %s %d', 'Hello', 'World', 2024);
 │ %Hello World 2024                            │
 └──────────────────────────────────────────────┘
 ```
+
+## overlay
+
+Replace the string `s` with the string `replace` starting from the 1-based `position` for `length` bytes. If `length` is omitted or negative, then it defaults to the length of `replace`.
```

Comment:
Firebird has an intuitive explanation: http://firebirdsql.org/refdocs/langrefupd21-intfunc-overlay.html

Slightly adjusted from their docs:

Replace a part of a string `s` with another string `replace`, starting at 1-based index `position`. By default, the number of characters removed from `s` equals the length of `replacement`. If `length` (the optional fourth argument) is specified, a different number of characters is removed.